core/indexer/factory/indexerFactory.go

```diff
@@ -99,11 +98,14 @@ func createElasticProcessor(args *ArgsIndexerFactory) (indexer.ElasticProcessor,
 }
 
 func checkDataIndexerParams(arguments *ArgsIndexerFactory) error {
+	if arguments.IndexerCacheSize < 0 {
+		return indexer.ErrNegativeCacheSize
+	}
 	if check.IfNil(arguments.AddressPubkeyConverter) {
-		return fmt.Errorf("%w when setting AddressPubkeyConverter in indexer", indexer.ErrNilPubkeyConverter)
+		return indexer.ErrNilPubkeyConverter
```

Comment:
Since you are returning the same error in 2 different places this will be harder to debug. Suggestion to revert the code. See below comment for test usage.

Reply:
done.